src/gsi/setupgnssrspd.f90

```diff
@@ -0,0 +1,951 @@
+!   language: f90
+!   machine:  ibm RS/6000 SP
+!
+!$$$
+  use mpeu_util, only: die,perr,getindex
+  use kinds, only: r_kind,r_single,r_double,i_kind
+  use m_obsdiagNode, only: obs_diag
+  use m_obsdiagNode, only: obs_diags
+  use m_obsdiagNode, only: obsdiagLList_nextNode
+  use m_obsdiagNode, only: obsdiagNode_set
+  use m_obsdiagNode, only: obsdiagNode_get
+  use m_obsdiagNode, only: obsdiagNode_assert
+
+  use obsmod, only: rmiss_single,&
+                    lobsdiagsave,nobskeep,lobsdiag_allocated,time_offset,&
+                    lobsdiag_forenkf,aircraft_recon
+  use obsmod, only: netcdf_diag, binary_diag, dirname, ianldate
+  use nc_diag_write_mod, only: nc_diag_init, nc_diag_header, nc_diag_metadata, &
+       nc_diag_write, nc_diag_data2d
+  use nc_diag_read_mod, only: nc_diag_read_init, nc_diag_read_get_dim, nc_diag_read_close
+  use m_obsNode, only: obsNode
+  use m_gnssrspdNode, only: gnssrspdNode
+  use m_gnssrspdNode, only: gnssrspdNode_appendto
+  use m_obsLList, only: obsLList
+  use obsmod, only: luse_obsdiag
+  use gsi_4dvar, only: nobs_bins,hr_obsbin
+  use guess_grids, only: nfldsig,hrdifsig,ges_lnprsl, &
+           comp_fact10,sfcmod_gfs,sfcmod_mm5
+  use guess_grids, only: geop_hgtl
+  use gridmod, only: nsig,get_ij,twodvar_regional
+  use qcmod, only: npres_print,ptop,pbot
+  use constants, only: one,grav,rd,zero,four,tiny_r_kind, &
+       half,two,cg_term,huge_single,r1000,wgtlim
+  use jfunc, only: jiter,last,miter,jiterstart
+  use state_vectors, only: svars3d, levels
+  use qcmod, only: dfact,dfact1
+  use convinfo, only: nconvtype,cermin,cermax,cgross,cvar_b,cvar_pg,ictype
+  use convinfo, only: icsubtype
+  use gsi_bundlemod, only : gsi_bundlegetpointer
+  use gsi_metguess_mod, only : gsi_metguess_get,gsi_metguess_bundle
+  use m_dtime, only: dtime_setup, dtime_check
+  use sparsearr, only: sparr2, new, size, writearray, fullarray
+
+  ! The following variables are the coefficients that describe the
+  ! linear regression fits that are used to define the dynamic
+  ! observation error (DOE) specifications for all reconnissance
+  ! observations collected within hurricanes/tropical cyclones; these
+  ! apply only to the regional forecast models (e.g., HWRF); Henry
+  ! R. Winterbottom (henry.winterbottom@noaa.gov).
+  
+  use obsmod, only: uv_doe_a_213,uv_doe_b_213
+  
+  implicit none
+
+! Declare passed variables
+  type(obsLList ),target,dimension(:),intent(in):: obsLL
+  type(obs_diags),target,dimension(:),intent(in):: odiagLL
+
+  logical                                          ,intent(in   ) :: conv_diagsave
+  integer(i_kind)                                  ,intent(in   ) :: lunin,mype,nele,nobs
+  real(r_kind),dimension(100+7*nsig)               ,intent(inout) :: awork
+  real(r_kind),dimension(npres_print,nconvtype,5,3),intent(inout) :: bwork
+  integer(i_kind)                                  ,intent(in   ) :: is ! ndat index
+
+! Declare local variables
+  real(r_kind),parameter:: ten=10.0_r_kind
```

Comment:
`constants.f90` defines `ten`.  Remove this line and add `ten` to the `use constants` line above.

Reply:
Thanks for the suggestion!